### PR TITLE
Add functions from base/reducedim.jl to doc/base/src/arrays.md

### PR DESCRIPTION
**Pull request by thchr**
This follows up on https://github.com/JuliaLang/julia/pull/29604#issuecomment-429105515: several of the doc-string'ed, exported functions (with `AbstractArray` inputs) from `base/reducedim.jl` were not included in the documentation, leading to doc-tests not being run.

I think the reasoning behind the omissions is that identically-named functions are already listed in the section on iterable collections in `doc/src/base/collections.md`---but those functions apply to generic iterables, so the `AbstractArray`-signature variants are not included. One of the downside of this is that doc-tests are not run for the `AbstractArray` variants.

So, in this PR, I include all the doc-string'ed, exported functions from `base/reducedim.jl` in the `arrays.md` documentation. In addition, I moved functions from the `collections.md` documentation to the `arrays.md` documentation if their type-inputs only allow `AbstractArray` (`maximum!`, `minimum!`, `findmin!`, `findmax!`, `any!`, `all!`, `sum!`, `prod!`, ). This latter move is perhaps not a net positive, so please let me know if that seems meaningful or not.

More broadly, let me know if the additions to `arrays.md` should go elsewhere (files, subsections, etc.)

### CODE REVIEW

**Comment by thchr** [comment]
@fredrikekre : I wonder if I could request a review from you, given https://github.com/JuliaLang/julia/pull/29604#issuecomment-429225123.

**Comment by thchr** [comment]
This is obviously not a crucial PR, but I thought I'd bump it anyway as I guess it is not likely to be a particularly controversial one either.

File: doc/src/base/arrays.md [hunk 1]
```diff
@@ -136,6 +136,20 @@ Base.findnext(::Any, ::Integer)
 Base.findnext(::Function, ::Any, ::Integer)
 Base.findprev(::Any, ::Integer)
 Base.findprev(::Function, ::Any, ::Integer)
+Base.findmin(::AbstractArray)
```
Comment (fredrikekre) [inline doc/src/base/arrays.md]: This method (and some others that you moved) work on e.g. tuples as well, but is this another docstring then?

**Review by fredrikekre — COMMENTED** [review]
Sorry, forgot about this one.

File: doc/src/base/arrays.md (hunk 1 above)
Comment (thchr) [reply doc/src/base/arrays.md]: I added `findmin` and moved `findmin!`: the former indeed has other methods, but for the latter, I can only find a single method signature taking `(::AbstractArray, ::AbstractArray, ::AbstractArray)`. 
The methods I removed from `collections.md` (and then readded to `arrays.md`) only had `::AbstractArray`-type signature variants.
Thanks for finding time to review this!

**Comment by ViralBShah** [comment]
I did a spot check on many of these, and it seems like they are now in the documentation. Please reopen if not the case, or a new PR if some things are still missing.

**Pull request closed without merging**